dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -54,641 +46,238 @@
-
-    /**
-     * Executes a SQL query that will return all the Vanity URLs that belong to a specific Site. This method moved from
-     * using the ES index to using a SQL query in order to avoid situations where the index was not fully updated when
-     * reading new data.
-     *
-     * @param siteId            The Identifier of the Site whose Vanity URLs will be retrieved.
-     * @param languageId        The language ID used to created the Vanity URLs.
-     * @param includeSystemHost If set to {@code true}, the results will include Vanity URLs that were created under
-     *                          System Host. Otherwise, set to {@code false}.
-     *
-     * @return The list of Vanity URLs.
-     */
-    @CloseDBIfOpened
-    private List<Contentlet> searchAndPopulate(final String siteId, final Long languageId, final boolean
-            includeSystemHost) {
-        List<Contentlet> contentlets = new ArrayList<>();
-        final StringBuilder query = new StringBuilder();
-        query.append("SELECT cvi.live_inode FROM contentlet c ");
-        query.append("INNER JOIN identifier i ON c.identifier = i.id AND i.host_inode ");
-        if (includeSystemHost) {
-            query.append("IN ('" + Host.SYSTEM_HOST + "', ?) ");
-        } else {
-            query.append("= ? ");
-        }
-        query.append("INNER JOIN contentlet_version_info cvi ON c.identifier = cvi.identifier AND c.inode = cvi" +
-                ".live_inode ");
-        query.append("WHERE c.language_id = ? AND c.structure_inode IN (SELECT s.inode FROM structure s WHERE s" +
-                ".structuretype = ?)");
-        try {
-            final List<Map<String, Object>> vanityUrls = new DotConnect().setSQL(query.toString())
-                    .addParam(siteId)
-                    .addParam(languageId)
-                    .addParam(BaseContentType.VANITY_URL.getType())
-                    .loadObjectResults();
-            final List<String> vanityUrlInodes = vanityUrls.stream().map(vanity -> vanity.get("live_inode").toString
-                    ()).collect(Collectors.toList());
-            contentlets = this.contentletAPI.findContentlets(vanityUrlInodes);
-        } catch (final DotDataException e) {
-            Logger.error(this, String.format("An error occurred when retrieving Vanity URLs: siteId=[%s], " +
-                    "languageId=[%s], includeSystemHost=[%s]", siteId, languageId, includeSystemHost), e);
-        } catch (final DotSecurityException e) {
-            Logger.error(this, String.format("An error occurred when retrieving Vanity URLs: siteId=[%s], " +
-                    "languageId=[%s], includeSystemHost=[%s]", siteId, languageId, includeSystemHost), e);
-        }
-        return contentlets;
+  @Override
+  public void populateAllVanityURLsCache() throws DotDataException {
+    for (Host host : Try.of(() -> APILocator.getHostAPI().findAllFromDB(APILocator.systemUser(), false)).getOrElse(ImmutableList.of())) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'host' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)